### PR TITLE
Cookie initialy baked by NetworkToCode Cookie Drift Manager Tool

### PR DESCRIPTION
**Pull request by snaselj**
Template:

```
{
    "template": "https://github.com/nautobot/cookiecutter-nautobot-app.git",
    "dir": "nautobot-app",
    "ref": "develop",
    "path": null
}
```

Cookie:

```
{
    "remote": "https://github.com/nautobot/nautobot-app-version-control.git",
    "path": "/opt/ntc/drift-manager/outputs/nautobot-app-version-control",
    "repository_path": "/opt/ntc/drift-manager/outputs/nautobot-app-version-control",
    "dir": "",
    "branch_prefix": "drift-manager",
    "context": {
        "codeowner_github_usernames": "@smith-ntc",
        "full_name": "Network to Code, LLC",
        "email": "opensource@networktocode.com",
        "github_org": "nautobot",
        "app_name": "nautobot_version_control",
        "verbose_name": "Nautobot Version Control",
        "app_slug": "nautobot-version-control",
        "project_slug": "nautobot-app-version-control",
        "repo_url": "https://github.com/nautobot/nautobot-app-version-control",
        "base_url": "version-control",
        "min_nautobot_version": "2.0.3",
        "max_nautobot_version": "2.9999",
        "camel_name": "NautobotVersionControl",
        "project_short_description": "Nautobot Version Control with Dolt",
        "model_class_name": "None",
        "open_source_license": "Apache-2.0",
        "docs_base_url": "https://docs.nautobot.com",
        "docs_app_url": "https://docs.nautobot.com/projects/version-control/en/latest",
        "_template": "https://github.com/nautobot/cookiecutter-nautobot-app.git",
        "_output_dir": "/opt/ntc/drift-manager/outputs",
        "_repo_dir": "/opt/ntc/drift-manager/outputs/.cookiecutters/cookiecutter-nautobot-app/nautobot-app",
        "_checkout": "develop"
    },
    "base_branch": "develop",
    "remote_name": "origin",
    "pull_request_strategy": "PullRequestStrategy.CREATE",
    "post_actions": [
        "PostAction.BLACK"
    ],
    "baked_commit_ref": "",
    "draft": true
}
```

CLI Arguments:

```
{
    "cookie_dir": "",
    "input": false,
    "json_filename": "json/version-control.json",
    "output_dir": "./outputs",
    "push": true,
    "template": "https://github.com/nautobot/cookiecutter-nautobot-app.git",
    "template_dir": "nautobot-app",
    "template_ref": "develop",
    "pull_request": null,
    "post_action": [
        "black"
    ],
    "disable_post_actions": false,
    "draft": true
}
```

### CODE REVIEW

**Comment by snaselj** [comment]
This pull request has been cleaned up and is ready for the review. @itdependsnetworks 

- Bumped Nautobot dependency to `v2.0.3`.
- Ignored linter issues.

**Comment by itdependsnetworks** [comment]
I think this is good...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,41 +136,46 @@ jobs:
...
   unittest:
     needs:
-      - "pylint"
+      - "check-in-docker"
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.11"]
+        db-backend: ["postgresql"]
```
Comment (itdependsnetworks) [inline .github/workflows/ci.yml]: This is not going to be needed, along with anything else that is postres or mysql.

File: development/docker-compose.mysql.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
```
Comment (itdependsnetworks) [inline development/docker-compose.mysql.yml]: Can remove this file.

File: development/docker-compose.postgres.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (itdependsnetworks) [inline development/docker-compose.postgres.yml]: Can remove this file.

File: development/docker-compose.mysql.yml (hunk 2 above)
Comment (snaselj) [reply development/docker-compose.mysql.yml]: I see, however rather keep those compose files. Drift management would recreate it each time there will be any change.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (snaselj) [reply .github/workflows/ci.yml]: Removed

File: development/creds.example.env [hunk 4]
```diff
@@ -12,6 +13,14 @@ NAUTOBOT_SUPERUSER_EMAIL=admin@example.com
...
 NAUTOBOT_SUPERUSER_API_TOKEN=0123456789abcdef0123456789abcdef01234567
 
+# Postgres
+POSTGRES_PASSWORD=${NAUTOBOT_DB_PASSWORD}
+PGPASSWORD=${NAUTOBOT_DB_PASSWORD}
+
+# MySQL Credentials
+MYSQL_ROOT_PASSWORD=${NAUTOBOT_DB_PASSWORD}
+MYSQL_PASSWORD=${NAUTOBOT_DB_PASSWORD}
+
```
Comment (glennmatthews) [inline development/creds.example.env]: Remove as irrelevant to this App.

File: development/development.env [hunk 5]
```diff
@@ -21,11 +19,24 @@ NAUTOBOT_REDIS_PORT=6379
...
+
+# Needed for Postgres should match the values for Nautobot above
+POSTGRES_USER=${NAUTOBOT_DB_USER}
+POSTGRES_DB=${NAUTOBOT_DB_NAME}
+
+# Needed for MYSQL should match the values for Nautobot above
+MYSQL_USER=${NAUTOBOT_DB_USER}
+MYSQL_DATABASE=${NAUTOBOT_DB_NAME}
+MYSQL_ROOT_HOST=%
+
```
Comment (glennmatthews) [inline development/development.env]: Remove as irrelevant to this App

File: development/docker-compose.mysql.yml (hunk 2 above)
Comment (glennmatthews) [reply development/docker-compose.mysql.yml]: They're just noise though for this app. Can we make drift manager smarter?

File: development/development_mysql.env
Comment (glennmatthews) [inline development/development_mysql.env:1]: Remove, irrelevant to this App.

File: docs/admin/install.md [hunk 6]
```diff
@@ -67,31 +61,7 @@ Then restart (if necessary) the Nautobot services which may include:
 sudo systemctl restart nautobot nautobot-worker nautobot-scheduler
 ```
 
-## App Configuration
```
Comment (glennmatthews) [inline docs/admin/install.md:70]: Removing this section looks to have broken docs building in CI.

File: docs/dev/code_reference/package.md [hunk 7]
```diff
@@ -0,0 +1 @@
+::: nautobot_version_control
```
Comment (glennmatthews) [inline docs/dev/code_reference/package.md:1]: Adding this appears to have broken CI docs build.

File: mkdocs.yml [hunk 8]
```diff
@@ -115,5 +123,6 @@ nav:
       - Architecture Decision Records: "dev/arch_decision.md"
       - Code Reference:
           - "dev/code_reference/index.md"
+          - Package: "dev/code_reference/package.md"
           - API: "dev/code_reference/api.md"
```
Comment (glennmatthews) [inline mkdocs.yml]: Are these two files redundant or do they document separate things?

File: nautobot_version_control/__init__.py [hunk 9]
```diff
@@ -138,8 +143,8 @@ def query_registry(model, registry):
 
 
 def is_versioned_model(model):
-    """
-    Determines whether a model's is under version control.
+    """Determines whether a model's is under version control.
+
```
Comment (glennmatthews) [inline nautobot_version_control/__init__.py]: Ew, is Ruff enforcing this style? I'd hope that it would permit both styles, as both are valid Python docstrings.

File: nautobot_version_control/models.py [hunk 10]
```diff
@@ -411,6 +418,7 @@ class PullRequest(BaseModel):
     # can't create Foreign Key to dolt_branches table :(
     source_branch = models.CharField(max_length=1024)
     destination_branch = models.CharField(max_length=1024)
+    # pylint: disable=nb-string-field-blank-null
     description = models.TextField(blank=True, null=True)
```
Comment (glennmatthews) [inline nautobot_version_control/models.py]: Ideally should fix this rather than disabling the check.

File: nautobot_version_control/models.py [hunk 11]
```diff
@@ -524,6 +532,7 @@ class PullRequestReview(BaseModel):
     reviewer = models.ForeignKey(User, on_delete=CASCADE)
     reviewed_at = models.DateTimeField(auto_now_add=True, blank=True, null=True)
     state = models.IntegerField(choices=REVIEW_STATE_CHOICES, null=True)
+    # pylint: disable=nb-string-field-blank-null
     summary = models.TextField(blank=True, null=True)
```
Comment (glennmatthews) [inline nautobot_version_control/models.py]: Ideally should fix this rather than disabling the check.

File: pyproject.toml [hunk 12]
```diff
@@ -1,47 +1,49 @@
...
 ]
 
 [tool.poetry.dependencies]
 python = ">=3.8,<3.12"
-nautobot = "~1.5"
-# MySQL database adapter
+# Used for local development
+nautobot = "^2.0.3"
 mysqlclient = "^2.0.3"
-python-dotenv = "*"
```
Comment (glennmatthews) [inline pyproject.toml:32]: Are we confident that `python-dotenv` is not a dependency of this App?

File: tasks.py [hunk 13]
```diff
@@ -54,7 +56,7 @@ def is_truthy(arg):
             "compose_files": [
                 "docker-compose.base.yml",
                 "docker-compose.redis.yml",
-                "docker-compose.dolt-hosted.yml",
+                # "docker-compose.dolt-hosted.yml",
```
Comment (glennmatthews) [inline tasks.py]: why this change?

File: tasks.py [hunk 14]
```diff
@@ -425,27 +500,43 @@ def dbshell(context, db_name="", input_file="", output_file="", query=""):
...
     help={
+        "db-name": "Database name to create (default: Nautobot database)",
         "input-file": "SQL dump file to replace the existing database with. This can be generated using `invoke backup-db` (default: `dump.sql`).",
     }
 )
-def import_db(context, input_file="dump.sql"):
-    """Stop Nautobot containers and replace the current database with the dump into the running `db` container."""
-    docker_compose(context, "stop -- nautobot worker")
+def import_db(context, db_name="", input_file="dump.sql"):
+    """Stop Nautobot containers and replace the current database with the dump into `db` container."""
```
Comment (glennmatthews) [inline tasks.py:508]: Does this work at all with Dolt?

File: tasks.py [hunk 15]
```diff
@@ -468,25 +559,23 @@ def import_db(context, input_file="dump.sql"):
     }
 )
 def backup_db(context, db_name="", output_file="dump.sql", readable=True):
```
Comment (glennmatthews) [inline tasks.py:561]: Same question here.

File: pyproject.toml (hunk 12 above)
Comment (itdependsnetworks) [reply pyproject.toml:32]: that should be accurate, I remember removing it

File: tasks.py (hunk 13 above)
Comment (itdependsnetworks) [reply tasks.py]: This should be left in there.

**Comment by itdependsnetworks** [comment]
As @glennmatthews has mentioned, leaving the mysql / postgres will be counter productive. The issue is that it will leave the user with the understanding that this will work with one of them.

File: development/docker-compose.mysql.yml (hunk 2 above)
Comment (snaselj) [reply development/docker-compose.mysql.yml]: For sure, depends on @cmsirbu as the PO. This is more about customizing the template, the drift management process is fine here.

File: development/creds.example.env (hunk 4 above)
Comment (snaselj) [reply development/creds.example.env]: Would rather keep it here, and resolve this by customizing the template.

File: development/development.env (hunk 5 above)
Comment (snaselj) [reply development/development.env]: Would rather keep it here, and resolve this by customizing the template.

File: development/development_mysql.env
Comment (snaselj) [reply development/development_mysql.env:1]: Would rather keep it here, and resolve this by customizing the template.

File: pyproject.toml (hunk 12 above)
Comment (snaselj) [reply pyproject.toml:32]: We don't need `python-dotenv` now.

File: tasks.py (hunk 13 above)
Comment (snaselj) [reply tasks.py]: Removed to have the default configuration local (non-hosted) only.

**Comment by snaselj** [comment]
> As @glennmatthews has mentioned, leaving the mysql / postgres will be counter productive. The issue is that it will leave the user with the understanding that this will work with one of them.

My opinion is that we should rather customize the template over doing manual changes here. WDYT @cmsirbu @whitej6

File: tasks.py (hunk 14 above)
Comment (snaselj) [reply tasks.py:508]: Unfortunately not.
